### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by Farida-Hamid**
>In this project, I connected My React-Redux application to an existing API to create and remove books on a remote server.

### Steps:
- Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) for making API calls.
- Fetch book list from Bookstore API on Books page load.
- Refactor add book and remove book features to persist your changes in the server.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 import Categories from './components/Categories';
 import Header from './components/Header';
+import { recieveBooks } from './redux/books/books';
 
 function App() {
   const state = useSelector((state) => state);
+  const dispatch = useDispatch();
+  useEffect(() => async () => {
+    await dispatch(recieveBooks());
+  }, []);
```
Comment (thecodechaser) [inline src/App.jsx]: - [x] As requested by the previous reviewer, Please remove `async` and `await` from this hook, You should not use these with `redux-thunk`. Simply remove `async` and `await` from the `useEffect` hook.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -13,28 +16,66 @@ export const remove = (index) => ({
...
+        bookList.push({
+          item_id: key,
+          title: books[key][0].title,
+          author: books[key][0].author,
+          category: books[key][0].category,
+        });
+      });
+      dispatch(read(bookList));
+    });
+};
```
Comment (thecodechaser) [inline src/redux/books/books.js]: - [x] As requested by the previous reviewer, Please use `createAsyncThunk` to make API calls. Read this [comment](https://github.com/Farida-Hamid/Bookstore/pull/4#discussion_r975515299)

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Farida-Hamid👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Farida-Hamid,** 

### **Great work on making the changes 👏** 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## **Suggested changes**
_Check the comments under the review._

**- Hello I want to tell you that you made a good job and I see that you are struggling with `redux-toolkit`  but as requested by my colleague TSE the `createAsyncThunk` is required for the project I have attached some links that will help you finish the app quickly good luck.**


- [CreateAsyncThunk React/redux-tool tutorial](https://redux-toolkit.js.org/api/createAsyncThunk).

- [Video tutorial of React/redux-tool, createAsyncThunk](https://www.youtube.com/watch?v=SSy8CuJmh1o).

Capture d’écran 2022-09-23 102741

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Farida-Hamid   

Your project is complete! Great job for making the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Validations correctly implemented
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**